src/subclass/cell_renderer.rs

```diff
@@ -0,0 +1,658 @@
+                    minimum_size.as_mut_ptr(),
+                    natural_size.as_mut_ptr(),
+                );
+                let minimum_size = minimum_size.assume_init();
+                let natural_size = natural_size.assume_init();
+                (minimum_size, natural_size)
+            } else {
+                (-1, -1)
+            }
+        }
+    }
+    fn parent_get_preferred_height<P: IsA<Widget>>(
+        &self,
+        renderer: &CellRenderer,
+        widget: &P,
+    ) -> (i32, i32) {
+        unsafe {
+            let data = self.get_type_data();
+            let parent_class =
+                data.as_ref().get_parent_class() as *mut gtk_sys::GtkCellRendererClass;
+            if let Some(f) = (*parent_class).get_preferred_height {
+                let mut minimum_size = mem::MaybeUninit::uninit();
+                let mut natural_size = mem::MaybeUninit::uninit();
+                f(
+                    renderer.to_glib_none().0,
+                    widget.as_ref().to_glib_none().0,
+                    minimum_size.as_mut_ptr(),
+                    natural_size.as_mut_ptr(),
+                );
+                let minimum_size = minimum_size.assume_init();
+                let natural_size = natural_size.assume_init();
+                (minimum_size, natural_size)
+            } else {
+                (-1, -1)
+            }
+        }
+    }
+    fn parent_get_preferred_height_for_width<P: IsA<Widget>>(
+        &self,
+        renderer: &CellRenderer,
+        widget: &P,
+        width: i32,
+    ) -> (i32, i32) {
+        unsafe {
+            let data = self.get_type_data();
+            let parent_class =
+                data.as_ref().get_parent_class() as *mut gtk_sys::GtkCellRendererClass;
+            if let Some(f) = (*parent_class).get_preferred_height_for_width {
+                let mut minimum_size = mem::MaybeUninit::uninit();
+                let mut natural_size = mem::MaybeUninit::uninit();
+                f(
+                    renderer.to_glib_none().0,
+                    widget.as_ref().to_glib_none().0,
+                    width,
+                    minimum_size.as_mut_ptr(),
+                    natural_size.as_mut_ptr(),
+                );
+                let minimum_size = minimum_size.assume_init();
+                let natural_size = natural_size.assume_init();
+                (minimum_size, natural_size)
+            } else {
+                (-1, -1)
+            }
+        }
+    }
+
+    fn parent_get_aligned_area<P: IsA<Widget>>(
+        &self,
+        renderer: &CellRenderer,
+        widget: &P,
+        flags: CellRendererState,
+        cell_area: &gdk::Rectangle,
+    ) -> gdk::Rectangle {
+        unsafe {
+            let data = self.get_type_data();
+            let parent_class =
+                data.as_ref().get_parent_class() as *mut gtk_sys::GtkCellRendererClass;
+            let mut aligned_area = gdk::Rectangle::uninitialized();
```

Comment:
The correct pattern for this nowadays (we should deprecated `::uninitialize()`...) is

```rust
let mut aligned_area = mem::MaybeUninitialized::uninit();
...
    aligned_area.as_mut_ptr(),
...

from_glib_none(aligned_area)
```

Reply:
ok

Reply:
It actually doesn't compile as is:
````
   Compiling gtk v0.8.0 (/home/hub/source/gtk-rs)
error[E0277]: the trait bound `gdk_sys::GdkRectangle: glib::translate::Ptr` is not satisfied
    --> src/subclass/cell_renderer.rs:312:28
     |
312  |             from_glib_none(aligned_area.assume_init())
     |                            ^^^^^^^^^^^^^^^^^^^^^^^^^^ the trait `glib::translate::Ptr` is not implemented for `gdk_sys::GdkRectangle`
     | 
    ::: /home/hub/.cargo/git/checkouts/glib-928cf7b282977403/597cba3/src/translate.rs:1131:33
     |
1131 | pub unsafe fn from_glib_none<P: Ptr, T: FromGlibPtrNone<P>>(ptr: P) -> T {
     |                                 --- required by this bound in `glib::translate::from_glib_none`

error[E0277]: the trait bound `gdk::Rectangle: glib::translate::FromGlibPtrNone<gdk_sys::GdkRectangle>` is not satisfied
    --> src/subclass/cell_renderer.rs:312:13
     |
312  |             from_glib_none(aligned_area.assume_init())
     |             ^^^^^^^^^^^^^^ the trait `glib::translate::FromGlibPtrNone<gdk_sys::GdkRectangle>` is not implemented for `gdk::Rectangle`
     | 
    ::: /home/hub/.cargo/git/checkouts/glib-928cf7b282977403/597cba3/src/translate.rs:1131:41
     |
1131 | pub unsafe fn from_glib_none<P: Ptr, T: FromGlibPtrNone<P>>(ptr: P) -> T {
     |                                         ------------------ required by this bound in `glib::translate::from_glib_none`
     |
     = help: the following implementations were found:
               <gdk::Rectangle as glib::translate::FromGlibPtrNone<*const gdk_sys::GdkRectangle>>
               <gdk::Rectangle as glib::translate::FromGlibPtrNone<*mut gdk_sys::GdkRectangle>>

error: aborting due to 2 previous errors
````

Reply:
`from_glib_none(&aligned_area.assume_init())` should work. It needs a pointer here.

Reply:
it doesn't.

Reply:
Ok, I'll take a look in a bit once you updated your branch for the other things :)

Reply:
`from_glib_none(&aligned_area.assume_init() as *const _)` would be the solution. But as the autogenerated code still uses `uninitialized()`, let's keep it as is for now for consistency.

We'll change this later once we got rid of `uninitialized()` in general and also changed the code generator accordingly.

Reply:
I wonder if we can't autogenerate this too...

Reply:
See https://github.com/gtk-rs/gtk/pull/937#discussion_r365506403 . A subset of these can potentially be autogenerated (there's an old PR against gir for this), yes.